--- cpp/Makefile.orig	2011-02-22 12:35:56.000000000 +0000
+++ cpp/Makefile	2011-02-22 12:36:51.000000000 +0000
@@ -1,3 +1,3 @@
 all:
-	g++ -g -Wall -O2 -I/gsc/pkg/bio/samtools/samtools-0.1.6/ BreakDancerMax.cpp AlnParser.cpp Poisson.cpp -o breakdancer_max -lm -lz -L/gsc/pkg/bio/samtools/samtools-0.1.6/ -lbam
+	g++ -g -Wall -O2 -I$(SAMTOOLSDIR)/include BreakDancerMax.cpp AlnParser.cpp Poisson.cpp -o breakdancer_max -lm -lz -L$(SAMTOOLSDIR)/lib -lbam
 
