--- Trinity.pl.orig	2013-11-10 22:42:41.000000000 +0000
+++ Trinity.pl	2014-04-04 10:14:30.641327001 +0100
@@ -660,7 +660,7 @@
 
 ## Check Java version:
 unless ($NO_RUN_BUTTERFLY_FLAG || $NO_RUN_CHRYSALIS_FLAG) {
-  my $java_version = `java -version 2>&1 `;
+  my $java_version = `java -Xmx64m -version 2>&1 `;
   unless ($java_version =~ /(java|openjdk) version \"1\.[67]\./) {
       die "Error, Trinity requires access to Java version 1.6 or 1.7.  Currently installed version is: $java_version";
   }
