--- inc/common.mk.orig	2012-02-07 19:19:02.000000000 +0000
+++ inc/common.mk	2012-08-16 15:12:09.000000000 +0100
@@ -145,16 +145,16 @@
 CFLAGS += ${HG_WARN}
 
 ifeq (${SCRIPTS},)
-    SCRIPTS=${HOME}/bin/scripts
+    SCRIPTS=${PREFIX}/bin/scripts
 endif
 ifeq (${CGI_BIN},)
-    CGI_BIN=/usr/local/apache/cgi-bin
+    CGI_BIN=${PREFIX}/apache/cgi-bin
 endif
 ifeq (${DOCUMENTROOT},)
-    DOCUMENTROOT=/usr/local/apache/htdocs
+    DOCUMENTROOT=${PREFIX}/apache/htdocs
 endif
 ifeq (${BINDIR},)
-    BINDIR = ${HOME}/bin/${MACHTYPE}
+    BINDIR = ${PREFIX}/bin/${MACHTYPE}
 endif
 ifeq (${ENCODE_PIPELINE_BIN},)
     ENCODE_PIPELINE_BIN=/cluster/data/encode/pipeline/bin
