--- makefile.orig       2012-10-10 10:11:12.000000000 +0100
+++ makefile    2012-10-10 10:11:29.000000000 +0100
@@ -1,6 +1,6 @@
 DEFINE =
 CXX = g++
-CXXFLAGS = -fomit-frame-pointer -O3 -ffast-math -funroll-loops -mmmx -msse -msse2 -msse3 -fmessage-length=0  #-MMD -MP -MF #-g3 -Wall -maccumulate-outgoing-args
+CXXFLAGS = -fomit-frame-pointer -O3 -ffast-math -funroll-loops -mmmx -msse -msse2 -msse3 -fmessage-length=0 -I$(BOOSTINCLUDE) #-MMD -MP -MF #-g3 -Wall -maccumulate-outgoing-args
 
 LFLAGS = -lz
