--- Makefile.orig	2015-06-19 15:56:52.895453123 +0000
+++ Makefile	2015-06-19 15:57:42.611455127 +0000
@@ -20,9 +20,9 @@
 endif
 
 CFLAGS=-Wall -O2
-BLASFLAGS=-L/usr/lib64/atlas -llapack -lcblas -latlas
+BLASFLAGS=-L$(ATLASLIB) -lsatlas
 LINKFLAGS=-lm -lpthread -ldl
-ZLIB=zlib-1.2.8/libz.so.1.2.8
+ZLIB=zlib-1.2.8/libz.a
 
 ifeq ($(SYS), MAC)
   GCC_GTEQ_43 := $(shell expr `g++ -dumpversion | sed -e 's/\.\([0-9][0-9]\)/\1/g' -e 's/\.\([0-9]\)/0\1/g' -e 's/^[0-9]\{3,4\}$$/&00/'` \>= 40300)
