--- t/output.t~	2013-12-03 12:30:15.290265577 +0000
+++ t/output.t	2013-12-03 12:47:19.753297305 +0000
@@ -14,9 +14,9 @@
 my $removeRun = $ARGV[1] // 1;
 
 #program locations
-my $blastDirectory = '/usr/bin/';
-my $mummerDirectory = '/usr/bin/';
-my $muscleExecutable = '/usr/bin/muscle';
+my $blastDirectory = "$ENV{NCBIBLASTBIN}/";
+my $mummerDirectory = "$ENV{MUMMERBIN}/";
+my $muscleExecutable = "$ENV{MUSCLEBIN}/muscle";
 
 #get script location via File::Basename
 my $SCRIPT_LOCATION = dirname(__FILE__);
@@ -241,7 +241,7 @@
 sub _runPanseq{
 	my $t=shift;
 	
-	my $systemLine="perl $SCRIPT_LOCATION/../lib/panseq.pl $SCRIPT_LOCATION/$t.batch";
+	my $systemLine="perl $ENV{PANSEQBIN}/panseq.pl $SCRIPT_LOCATION/$t.batch";
 	print "Systemline: $systemLine\n";
 	system($systemLine);
 }
