--- Makefile.alces	2016-05-06 12:30:09.798504760 +0100
+++ Makefile	2016-05-06 12:30:45.161505486 +0100
@@ -1,4 +1,4 @@
-CUDAPATH=/usr/local/cuda
+CUDAPATH=${CUDA_HOME}
 
 # Have this point to an old enough gcc (for nvcc)
 GCCPATH=/usr
@@ -9,4 +9,4 @@
 drv:
 	PATH=.:${CCPATH}:${PATH} ${NVCC} -arch=compute_20 -ptx compare.cu -o compare.ptx
 	g++ -O3 -I${CUDAPATH}/include -c gpu_burn-drv.cpp
-	g++ -o gpu_burn gpu_burn-drv.o -O3 -lcuda -L${CUDAPATH}/lib64 -L${CUDAPATH}/lib -Wl,-rpath=${CUDAPATH}/lib64 -Wl,-rpath=${CUDAPATH}/lib -lcublas -lcudart -o gpu_burn
+	g++ -o gpu_burn gpu_burn-drv.o -O3 -lcuda -L${CUDAPATH}/lib64 -L${CUDAPATH}/lib64/stubs -L${CUDAPATH}/lib -Wl,-rpath=${CUDAPATH}/lib64 -Wl,-rpath=${CUDAPATH}/lib64/stubs -Wl,-rpath=${CUDAPATH}/lib -lcublas -lcudart -o gpu_burn
