# trinity(7) -- Trinity v2.2.0 (GridScheduler)

## DESCRIPTION

Trinity, developed at the Broad Institute and the Hebrew University
of Jerusalem, represents a novel method for the efficient and robust
de novo reconstruction of transcriptomes from RNA-seq data. Trinity
combines three independent software modules: Inchworm, Chrysalis,
and Butterfly, applied sequentially to process large volumes of
RNA-seq reads. Trinity partitions the sequence data into many
individual de Bruijn graphs, each representing the transcriptional
complexity at at a given gene or locus, and then processes each
graph independently to extract full-length splicing isoforms and to
tease apart transcripts derived from paralogous genes.

For more information on running Trinity, please visit:
  * <http://trinityrnaseq.github.io/>

## LICENSE

This work is licensed under a Creative Commons Attribution-ShareAlike
4.0 International License.

See <http://creativecommons.org/licenses/by-sa/4.0/> for more
information.

## COPYRIGHT

Copyright (C) 2016 Alces Software Ltd.

