--- universe_wsgi.ini.sample	2013-04-11 16:37:53.235478451 +0100
+++ universe_wsgi.ini	2013-04-11 16:39:15.433505304 +0100
@@ -376,7 +376,7 @@
 # also causes the files used by PBS/SGE (submission script, output, and error)
 # to remain on disk after the job is complete.  Debug mode is disabled if
 # commented, but is uncommented by default in the sample config.
-debug = True
+debug = False
 
 # Check for WSGI compliance.
 #use_lint = False
@@ -389,7 +389,7 @@
 
 # Enable live debugging in your browser.  This should NEVER be enabled on a
 # public site.  Enabled in the sample config for development.
-use_interactive = True
+use_interactive = False
 
 # Write thread status periodically to 'heartbeat.log',  (careful, uses disk
 # space rapidly!).  Useful to determine why your processes may be consuming a
