--- lib/galaxy/jobs/runners/drmaa.py.orig	2013-04-08 17:28:46.000000000 +0100
+++ lib/galaxy/jobs/runners/drmaa.py	2013-04-23 18:22:47.614371451 +0100
@@ -44,19 +44,46 @@
 #  - move to the job wrapper's working directory
 #  - execute the command
 #  - take the command's exit code ($?) and write it to a file.
-drm_template = """#!/bin/sh
-GALAXY_LIB="%s"
-if [ "$GALAXY_LIB" != "None" ]; then
-    if [ -n "$PYTHONPATH" ]; then
+drm_template = """#!/bin/bash
+if [ -f /opt/alces/etc/profile.d/alces-symphony.sh ]; then
+  . /opt/alces/etc/profile.d/alces-symphony.sh
+elif [ -f /etc/profile.d/alces-profiles.sh ]; then
+  . /etc/profile.d/alces-profiles.sh
+elif [ -f /etc/profile.d/modules.sh ]; then
+  . /etc/profile.d/modules.sh
+else
+  false
+fi
+if [ $? == 0 ]; then
+  ALCES_MODULES_LOAD=$(mktemp)
+  GALAXY_MODULE=%GALAXY_MODULE%
+  if module load $GALAXY_MODULE &>$ALCES_MODULES_LOAD; echo ":$LOADEDMODULES:" | grep -q ":$GALAXY_MODULE:"; then
+    rm -f $ALCES_MODULES_LOAD
+    if [ -f "$GALAXYDIR/etc/local-env.sh" ]; then
+      . $GALAXYDIR/etc/local-env.sh
+    fi
+    GALAXY_LIB="%s"
+    if [ "$GALAXY_LIB" != "None" ]; then
+      if [ -n "$PYTHONPATH" ]; then
         PYTHONPATH="$GALAXY_LIB:$PYTHONPATH"
-    else
+      else
         PYTHONPATH="$GALAXY_LIB"
+      fi
+      export PYTHONPATH
     fi
-    export PYTHONPATH
+    %s
+    cd %s
+    %s
+  else
+    echo "Failed to load modulefile: $GALAXY_MODULE" >>/dev/stderr
+    cat $ALCES_MODULES_LOAD >>/dev/stderr
+    rm -f $ALCES_MODULES_LOAD
+    false
+  fi
+else
+  echo "Unable to locate environment modules initialization script." >>/dev/stderr
+  false
 fi
-%s
-cd %s
-%s
 echo $? > %s
 """
 
@@ -133,6 +160,8 @@
         jt.jobName = ajs.job_name
         jt.outputPath = ":%s" % ajs.output_file
         jt.errorPath = ":%s" % ajs.error_file
+        # override any Gridscheduler-configured merging of stdout/err
+        jt.joinFiles = False
 
         # Avoid a jt.exitCodePath for now - it's only used when finishing.
         native_spec = job_destination.params.get('nativeSpecification', None)
