--- setup.py.orig	2016-01-23 22:16:11.000000000 +0000
+++ setup.py	2016-07-18 10:57:49.613624378 +0100
@@ -26,6 +26,7 @@
                   "msproteomicstoolslib.algorithms.alignment",
                   "msproteomicstoolslib.algorithms.shared",
                   "msproteomicstoolslib.algorithms.PADS",
+                  "msproteomicstoolslib.algorithms.graphs",
                   "msproteomicstoolslib.data_structures",
                   "msproteomicstoolslib.format",
                   "msproteomicstoolslib.math",
