--- configure.orig	2015-09-01 11:14:49.000000000 +0100
+++ configure	2016-03-09 12:43:06.447601194 +0000
@@ -15846,7 +15846,7 @@
 #Check for FLTK
 # as g++ seems needed and not gcc for compilation,
 #just check for presence of fltk-config program
-fltkfile=`pwd`/bin/fltk-config
+fltkfile=$(which fltk-config)
 as_ac_File=`$as_echo "ac_cv_file_$fltkfile" | $as_tr_sh`
 { $as_echo "$as_me:$LINENO: checking for $fltkfile" >&5
 $as_echo_n "checking for $fltkfile... " >&6; }
--- src/displayer.h.orig	2015-06-05 09:35:30.000000000 +0100
+++ src/displayer.h	2016-03-09 12:51:50.506601194 +0000
@@ -29,18 +29,18 @@
 #include <src/time.h>
 #include <src/args.h>
 
-#include <external/fltk-1.3.0/FL/Fl.H>
-#include <external/fltk-1.3.0/FL/Fl_Shared_Image.H>
-#include <external/fltk-1.3.0/FL/Fl_Double_Window.H>
-#include <external/fltk-1.3.0/FL/Fl_Scroll.H>
-#include <external/fltk-1.3.0/FL/Fl_Image.H>
-#include <external/fltk-1.3.0/FL/Fl_JPEG_Image.H>
-#include <external/fltk-1.3.0/FL/Fl_Box.H>
-#include <external/fltk-1.3.0/FL/fl_draw.H>
-#include <external/fltk-1.3.0/FL/Fl_Menu_Bar.H>
-#include <external/fltk-1.3.0/FL/Fl_File_Chooser.H>
-#include <external/fltk-1.3.0/FL/Fl_Float_Input.H>
-#include <external/fltk-1.3.0/FL/Fl_Text_Display.H>
+#include <FL/Fl.H>
+#include <FL/Fl_Shared_Image.H>
+#include <FL/Fl_Double_Window.H>
+#include <FL/Fl_Scroll.H>
+#include <FL/Fl_Image.H>
+#include <FL/Fl_JPEG_Image.H>
+#include <FL/Fl_Box.H>
+#include <FL/fl_draw.H>
+#include <FL/Fl_Menu_Bar.H>
+#include <FL/Fl_File_Chooser.H>
+#include <FL/Fl_Float_Input.H>
+#include <FL/Fl_Text_Display.H>
 
 #define GUI_BACKGROUND_COLOR (fl_rgb_color(240,240,240))
 #define GUI_INPUT_COLOR (fl_rgb_color(255,255,230))
--- src/manualpicker.h.orig	2015-06-05 09:35:30.000000000 +0100
+++ src/manualpicker.h	2016-03-09 12:52:02.118601194 +0000
@@ -24,18 +24,18 @@
 #include "src/args.h"
 #include "src/filename.h"
 #include "src/gui_entries.h"
-#include <external/fltk-1.3.0/FL/Fl.H>
-#include <external/fltk-1.3.0/FL/Fl_Shared_Image.H>
-#include <external/fltk-1.3.0/FL/Fl_Double_Window.H>
-#include <external/fltk-1.3.0/FL/Fl_Scroll.H>
-#include <external/fltk-1.3.0/FL/Fl_Image.H>
-#include <external/fltk-1.3.0/FL/Fl_JPEG_Image.H>
-#include <external/fltk-1.3.0/FL/Fl_Box.H>
-#include <external/fltk-1.3.0/FL/fl_draw.H>
-#include <external/fltk-1.3.0/FL/Fl_Menu_Bar.H>
-#include <external/fltk-1.3.0/FL/Fl_File_Chooser.H>
-#include <external/fltk-1.3.0/FL/Fl_Float_Input.H>
-#include <external/fltk-1.3.0/FL/Fl_Text_Display.H>
+#include <FL/Fl.H>
+#include <FL/Fl_Shared_Image.H>
+#include <FL/Fl_Double_Window.H>
+#include <FL/Fl_Scroll.H>
+#include <FL/Fl_Image.H>
+#include <FL/Fl_JPEG_Image.H>
+#include <FL/Fl_Box.H>
+#include <FL/fl_draw.H>
+#include <FL/Fl_Menu_Bar.H>
+#include <FL/Fl_File_Chooser.H>
+#include <FL/Fl_Float_Input.H>
+#include <FL/Fl_Text_Display.H>
 
 #define MWCOL1 300
 #define MWCOL2 60
