--- interproscan.sh.unmod	2016-09-23 13:50:20.000000000 +0100
+++ interproscan.sh	2017-01-20 11:51:22.117048571 +0000
@@ -12,6 +12,8 @@
 export EMBOSS_ACDROOT=bin/nucleotide
 export EMBOSS_DATA=bin/nucleotide
 
+export MALLOC_ARENA_MAX=1
+
 # Check Java is installed
 
 JAVA=$(type -p java)
@@ -62,6 +64,7 @@
 "$JAVA" \
 -XX:+UseParallelGC -XX:ParallelGCThreads=4 -XX:+AggressiveOpts \
 -XX:+UseFastAccessorMethods -Xms1028M -Xmx2048M \
+-XX:-UseCompressedClassPointers \
 -jar  interproscan-5.jar $@ -u $USER_DIR
 
 #end 
